### PR TITLE
Change the ByteString and Text parsers to handle '\r' newlines

### PR DESCRIPTION
**Pull request by mohsen3**

### CODE REVIEW

**Comment by ozataman** [comment]
Thank you for this. Could we add some small test files (and accompanying tests) with the different line endings that confirm that these parsers now work as intended?

**Comment by mohsen3** [comment]
@ozataman I just added a small test for the new case. It seems that it it working fine. I have also tested it manually by feeding in a Mac-Excel csv file that didn't work before.

File: src/Data/CSV/Conduit/Parser/ByteString.hs [hunk 1]
```diff
@@ -58,15 +58,17 @@ csv s = do
 row :: CSVSettings -> Parser (Maybe (Row ByteString))
 row csvs = csvrow csvs <|> badrow
 
+csvEndOfLine :: Parser ()
+csvEndOfLine = (word8 10 >> return ()) <|> (C8.string (B8.pack "\r\n") >> return ()) <|> (word8 13 >> return ())
```
Comment (mohsen3) [inline src/Data/CSV/Conduit/Parser/ByteString.hs]: Copied from
https://hackage.haskell.org/package/attoparsec-0.13.0.1/docs/src/Data-Attoparsec-ByteString-Internal.html
and modified to handle `\r` as well.

File: src/Data/CSV/Conduit/Parser/Text.hs [hunk 2]
```diff
@@ -58,15 +58,17 @@ csv s = do
 row :: CSVSettings -> Parser (Maybe (Row Text))
 row csvs = csvrow csvs <|> badrow
 
+csvEndOfLine :: Parser ()
+csvEndOfLine = (char '\n' >> return ()) <|> (string (T.pack "\r\n") >> return ()) <|> (char '\r' >> return ())
```
Comment (mohsen3) [inline src/Data/CSV/Conduit/Parser/Text.hs]: Copied from https://hackage.haskell.org/package/attoparsec-0.13.0.1/docs/src/Data-Attoparsec-Text-Internal.html
and modified to handle '\r'.

**Comment by dmvianna** [comment]
Merged in #56. I removed `"\r\n"` as there was no test for that.

**Pull request closed without merging**